homeassistant/components/overkiz/climate_entities/valve_heating_temperature_interface.py

```diff
@@ -0,0 +1,137 @@
+
+OVERKIZ_TO_HVAC_ACTION: dict[str, HVACAction] = {
+    OverkizCommandParam.OPEN: HVACAction.HEATING,
+    OverkizCommandParam.CLOSED: HVACAction.IDLE,
+}
+
+OVERKIZ_TO_PRESET_MODE: dict[str, str] = {
+    OverkizCommandParam.GEOFENCING_MODE: PRESET_NONE,
+    OverkizCommandParam.SUDDEN_DROP_MODE: PRESET_NONE,
+    OverkizCommandParam.AWAY: PRESET_AWAY,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.FROSTPROTECTION: PRESET_FROST_PROTECTION,
+    OverkizCommandParam.MANUAL: PRESET_MANUAL,
+}
+PRESET_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODE.items()}
+
+TEMPERATURE_SENSOR_DEVICE_INDEX = 2
+
+
+class ValveHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Valve Heating Temperature Interface device."""
+
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            TEMPERATURE_SENSOR_DEVICE_INDEX
+        )
+
+        self._attr_min_temp = cast(
+            float, self.executor.select_state(OverkizState.CORE_MIN_SETPOINT)
+        )
+        self._attr_max_temp = cast(
+            float, self.executor.select_state(OverkizState.CORE_MAX_SETPOINT)
+        )
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        return OVERKIZ_TO_HVAC_ACTION[
+            cast(str, self.executor.select_state(OverkizState.CORE_OPEN_CLOSED_VALVE))
+        ]
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature."""
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_TARGET_TEMPERATURE)
+        )
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if temperature := self.temperature_device.states[OverkizState.CORE_TEMPERATURE]:
+            return temperature.value_as_float
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_DEROGATION,
+            float(temperature),
+            OverkizCommandParam.FURTHER_NOTICE,
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        return
```

Comment:
A second thing that I don't really like here is that we have a compulsory "empty" function.

I suggested that we raise a HomeAssistantError in https://github.com/home-assistant/core/pull/86691/files#r1118390937 but now I wonder...
I'll ask for a second-opinion...

Reply:
Thanks! I fully agree and we have an empty function in another climate implementation as well. If we have a decision, I will fix that one (in a separate PR) as well.

Reply:
It seems empty function is OK in this case.